src/Microsoft.DotNet.Watcher.Core/Internal/Implementation/Project.cs

```diff
@@ -15,12 +17,27 @@ public Project(ProjectModel.Project runtimeProject)
             ProjectFile = runtimeProject.ProjectFilePath;
             ProjectDirectory = runtimeProject.ProjectDirectory;
 
-            Files = runtimeProject.Files.SourceFiles.Concat(
-                    runtimeProject.Files.ResourceFiles.Values.Concat(
-                    runtimeProject.Files.PreprocessSourceFiles.Concat(
-                    runtimeProject.Files.SharedFiles))).Concat(
-                    new string[] { runtimeProject.ProjectFilePath })
-                .ToList();
+            var compilerOptions = runtimeProject.GetCompilerOptions(targetFramework: null, configurationName: null);
```

Comment:
I think there are rare scenarios like that but I am not sure if we need to proactively address them now.